src/rprblender/engine/render_engine.py

```diff
@@ -145,7 +147,7 @@ def _render(self):
                 break
 
             render_iteration += 1
-            if render_iteration > 1 and self.render_update_samples < MAX_RENDER_ITERATIONS:
+            if render_iteration > 1 and self.render_update_samples < MAX_RENDER_ITERATIONS and not self.use_contour:
```

Comment:
move self.use_contour is true, other tests are unescessary.  Move first for logic?